patches/druntime/0007-Replace-buggy-time-based-profile-with-tick-based-pro.patch

```diff
@@ -0,0 +1,51 @@
+From 2ca0241a67ad67dc16f403ac6cb67563f9910388 Mon Sep 17 00:00:00 2001
+From: Stefan Koch <stefan.koch@sociomantic.com>
+Date: Fri, 26 Apr 2019 14:23:34 +0200
+Subject: [PATCH] Replace buggy time-based profile with tick based profile
+
+---
+ src/rt/trace.d | 13 ++++++-------
+ 1 file changed, 6 insertions(+), 7 deletions(-)
+
+diff --git a/src/rt/trace.d b/src/rt/trace.d
+index 07fd333..69afbb6 100644
+--- a/src/rt/trace.d
++++ b/src/rt/trace.d
+@@ -26,7 +26,7 @@ private
+ 
+ extern (C):
+ 
+-alias long timer_t;
++alias ulong timer_t;
+ 
+ /////////////////////////////////////
+ //
+@@ -292,7 +292,7 @@ private void trace_times(FILE* fplog, Symbol*[] psymbols)
+     // Print array
+     timer_t freq;
+     QueryPerformanceFrequency(&freq);
+-    fprintf(fplog,"\n======== Timer Is %lld Ticks/Sec, Times are in Microsecs ========\n\n",freq);
++    fprintf(fplog,"\n======== Times are in Ticks (Cycles) ========\n\n");
+     fprintf(fplog,"  Num          Tree        Func        Per\n");
+     fprintf(fplog,"  Calls        Time        Time        Call\n\n");
+     foreach (s; psymbols)
+@@ -313,12 +313,11 @@ private void trace_times(FILE* fplog, Symbol*[] psymbols)
+         if (calls == 0)
+             calls = 1;
+ 
+-        tl = (s.totaltime * 1000000) / freq;
+-        fl = (s.functime  * 1000000) / freq;
+-        percall = s.functime / calls;
+-        pl = (s.functime * 1000000) / calls / freq;
++        tl = s.totaltime;
++        fl = s.functime;
++        pl = s.functime / calls;
+ 
+-        fprintf(fplog,"%7llu%12lld%12lld%12lld     %.*s\n",
++        fprintf(fplog,"%llu %llu %llu %llu     %.*s\n",
+                       calls, tl, fl, pl, id.length, id.ptr);
+     }
```

Comment:
This is doing much more than what the title says, you should clarify why timings are being removed and why negative values were reported before.

Reply:
%lld -> print as signed number, %llu print as unsigned.
there is no timing removed the line that I removed was never used.

Reply:
OK, this is why the commit needs a better description, it is very hard to understand what is going on and what's your intention here. You are indeed removing the `freq` from this. You are also removing the space that makes the output looks like a table, now the header and the values won't be aligned anymore. Why did you do that?

I think having times in usec is useful to get a better grasp of how much real time is being spent. Why are you converting that into ticks? I still don't understand where is the issue you are trying to fix.

Reply:
ah the timing is not actually in usecs.
it's just ticks divded by a fixed constant.

Under windows it actually fetches a real processor frequency.
But under linux the frequency is hard-coded to 3 mhz.
Since it's hard to get the freq under linux I decided to go with ticks.

Reply:
All that should be explained in the commit message.